From a1879d1aef692886d9526b02eaa787a6535d4cdc Mon Sep 17 00:00:00 2001
From: Michal Vasko <mvasko@cesnet.cz>
Date: Tue, 12 May 2020 16:06:01 +0200
Subject: [PATCH] netopeer2: fix build errors and NACM username bug

Fix multiple declaration issues and NACM filtering bug for netopeer2 v1.1.27

Build Fixes:
- Fix multiple definition of 'some_msg' in CLI (commands.h, main.c)
- Fix multiple definition of 'np2_stderr_log' in log.h

NACM Fix:
- Get username based on nc session instead of sysrepo session
- Fixes #617 - https://github.com/CESNET/netopeer2/issues/617

---
 cli/commands.h     | 2 +-
 cli/main.c         | 1 +
 src/log.h          | 2 +-
 src/netconf_nmda.c | 2 +-
 4 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/cli/commands.h b/cli/commands.h
index b562c28..0252be3 100644
--- a/cli/commands.h
+++ b/cli/commands.h
@@ -17,7 +17,7 @@

 #include "cli_version.h"

-char some_msg[4096];
+extern char some_msg[4096];
 #define INSTRUCTION(format,args...) {snprintf(some_msg,4095,format,##args);printf("\n  %s",some_msg);}
 #define ERROR(function,format,args...) {snprintf(some_msg,4095,format,##args);fprintf(stderr,"%s: %s\n",function,some_msg);}

diff --git a/cli/main.c b/cli/main.c
index f017596..7c3db3a 100644
--- a/cli/main.c
+++ b/cli/main.c
@@ -36,6 +36,7 @@
 #include "linenoise/linenoise.h"

 int done;
+char some_msg[4096];

 extern char *config_editor;
 extern struct nc_session *session;
diff --git a/src/log.h b/src/log.h
index 0e42bff..66262e4 100644
--- a/src/log.h
+++ b/src/log.h
@@ -36,7 +36,7 @@ extern uint8_t np2_sr_verbose_level;
 /**
  * @brief netopeer2 flag whether to print messages to stderr (only if not daemon).
  */
-uint8_t np2_stderr_log;
+extern uint8_t np2_stderr_log;

 /**
  * @brief internal printing function, follows the levels from libnetconf2
diff --git a/src/netconf_nmda.c b/src/netconf_nmda.c
index aa42612..1f519bf 100644
--- a/src/netconf_nmda.c
+++ b/src/netconf_nmda.c
@@ -224,7 +224,7 @@ np2srv_rpc_getdata_cb(sr_session_ctx_t *session, const char *UNUSED(op_path), co
     ly_set_free(nodeset);

     /* perform correct NACM filtering */
-    ncac_check_data_read_filter(&data_get, sr_session_get_user(session));
+    ncac_check_data_read_filter(&data_get, np_get_nc_sess_user(session));

     /* add output */
     node = lyd_new_output_anydata(output, NULL, "data", data_get, LYD_ANYDATA_DATATREE);
--
2.25.1
